src/initialize.F90

```diff
@@ -109,10 +109,19 @@ subroutine initialize_run()
       ! Create linked lists for multiple instances of the same nuclide
       call same_nuclide_list()
 
-      ! Construct logarithmic energy grid for cross-sections
-      if (grid_method == GRID_LOGARITHM) then
+      ! Construct unionized or log energy grid for cross-sections
+      select case (grid_method)
+      case (GRID_NUCLIDE)
+        continue
+      case (GRID_MAT_UNION)
+        call time_unionize % start()
+!$omp critical
+        call unionized_grid()
+!$omp end critical
```

Comment:
This is not within an `!$omp parallel` region, so I don't see why a critical section is needed.